files/servicenow.rb

```diff
@@ -61,12 +61,14 @@ def parse_classification_fields(cmdb_record, classes_field, environment_field)
 
 # Abstract away the API calls.
 class ServiceNowRequest
-  def initialize(uri, http_verb, body, user, password)
+  def initialize(uri, http_verb, body, user, password, oauth_token)
+    raise ArgumentError, 'user/password or oauth_token must be specified' if oauth_token.nil? && (password.nil? || user.nil?)
```

Comment:
Took me a bit to figure out that your `if` statement here is correct. I think this would be simpler if it were written as
```
unless oauth_token || (user && password)
end
```